### PR TITLE
Use VSCode's python plugin to autoformat the python in amplipi

### PR DESCRIPTION
**Pull request by linknum23**
### Checklist

* [ ] Have you tested your changes and ensured they work?
* [ ] Does your submission pass linting & tests? You can test on localhost using `./scripts/test`

### CODE REVIEW

**Comment by rtertiaer** [comment]
not necessary, but I would 100% approve this PR if it also enabled failing CI if the linter says everything's not copacetic.

**Comment by rtertiaer** [comment]
I've squashed the test commit and fixups into oblivion and force-pushed this branch.

I've also swapped this to use pycodestyle for speed, and moved where it occurs in our CI pipeline for efficiency.

Finally, I wrapped up a ton of whitespace fixup changes, and a couple of `== None` => `is None` / `not Thing in Iterable` => `Thing not in Iterable` sort of changes.

File: hw/tests/preamp.py
Comment (linknum23) [inline hw/tests/preamp.py:12]: These files are all getting 4 space indented.

Comment (linknum23) [reply hw/tests/preamp.py:12]: 2 space should be the rule for all python in this repo

File: .github/workflows/python-app.yml
Comment (linknum23) [inline .github/workflows/python-app.yml:38]:
```suggestion
pycodestyle --ignore=E501,E722 --indent-size=2 scripts/ hw/ docs/
```

**Comment by rtertiaer** [comment]
@linknum23 If we're going to wholesale reformat giant swaths of the repo to fit one spacing standard, it should probably be the actual standard of 4 spaces and not 2 spaces.

**Comment by rtertiaer** [comment]
after one really messy rebase and a bug where autopep8 doesn't respect --indent-size=2, i'm considering this ship-able. getting this perfect in this PR is going to be hard for a codebase that's continually moving (god forbid I have to rebase again), and at this point I've spent a pretty inordinate amount of time on linting for some pretty questionable benefit. I'm pretty unwilling to dramatically rework any more of this PR but if there are small things I might put more time into it. :shrug:

**Review by linknum23 — COMMENTED** [review]
LGTM. I think the thing we are saving here is time in the future. This has potential to allow us to more easily review PRs.

**Review by klay2000 — COMMENTED** [review]
wow, that's a lot..
I appreciate the amount of frustration you went through to get this to behave the way we want it to!

@linknum23 in .vscode/settings.json there's a section that ignores a rule for autopep8, idk if you have any context for that. I was hoping to add autopep8 to extensions.json with the rules Ryan had set to be ignored in the settings but noticed that.

**Review by SteveMicroNova — COMMENTED** [review]
This PR seems to be solving a problem that I don't feel qualified to comment on as someone who didn't know the difference between Pascal, Snake, and Camel Casing until yesterday; I am not overly picky about coding styles in this fashion

My only comment is we've got to all remember that the lines you've now taken ownership as "last touched by" are last touched due to a reformatting and you don't strictly own any bugs, I've had some issues with gitblame due to being the one who reformatted the frontend some time ago

You have my approval, but again as someone who doesn't feel that I can have a proper informed opinion I am making a comment rather than a direct approval review

**Review by SteveMicroNova — APPROVED** [review]

**Review by klay2000 — APPROVED** [review]

**Pull request merged**
